FWCore/PluginManager/src/PluginManager.cc

```diff
@@ -245,7 +245,12 @@ namespace edmplugin {
           //TEMPORARY: to avoid possible deadlocks from ROOT, we must
           // take the lock ourselves
           R__LOCKGUARD2(gInterpreterMutex);
-          ptr.reset(new SharedLibrary(p));
+          try {
+            ptr = std::make_shared<SharedLibrary>(p);
+          } catch (cms::Exception const& e) {
+            throw cms::Exception("PluginLoadError")
```

Comment:
Instead of creating a new exception, how about just adding context
```C++
} catch(cms::Exception& e) {
   std::osstream s;
   s <<"While attempting to load plugin "<<iPlugin;
   e.addContext(s.str());
   throw;
}
```
In that way, if a non 'PluginLoadError' exception happened (or 'PluginLoadError' message were ever changed), we would preserve the original message.

Reply:
Thanks a lot, that is indeed much better! I will try that later.

Reply:
Hi, I implemented it with the added context now, almost exactly like you proposed. Thanks again!